### PR TITLE
feat: add VERS debian versioning scheme support

### LINKED ISSUE #55: feat: add VERS debian versioning scheme support
## Problem Statement

The VERS specification supports Debian versioning scheme, but go-univers currently only implements Debian ecosystem support without VERS integration. Users cannot use VERS range notation with Debian versions like `vers:debian/>=1:2.4.1-1|<1:3.0.0`.

## Research

From the [VERS specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst), Debian uses comparison operators "<<, <=, =, >=, and >>" and is notable for using "epochs to disambiguate versions". The [univers Python implementation](https://github.com/aboutcode-org/univers) includes Debian support (as "deb") in its VERS implementation.

Debian versions follow the format: `[epoch:]upstream-version[-debian-revision]` where epoch is used for version ordering disambiguation.

## Current State

- ✅ Debian ecosystem exists at `pkg/ecosystem/debian/`
- ✅ Debian version parsing and comparison implemented with epoch support
- ❌ No VERS integration for Debian versioning scheme
- ❌ Cannot parse/evaluate ranges like `vers:debian/>=1:2.4.1-1|<1:3.0.0`

## Implementation Strategy

Following the established pattern used for npm, pypi, go, and maven VERS schemes:

1. **Create `pkg/spec/vers/debian.go`** with:
   - `debianContains(constraints []string, version string) (bool, error)` function
   - `intervalToDebianRanges(interval interval) []string` function
   - Handle "debian" scheme name (though spec mentions "deb")

2. **Update `pkg/spec/vers/vers.go`**:
   - Add `"debian": debianContains` to `schemeToContains` map
   - Add `case "debian": rangeStrs = intervalToDebianRanges(interval)` to `toRanges` function
   - Update supported ecosystems documentation

3. **Add comprehensive tests** in `pkg/spec/vers/debian_test.go`

## Success Criteria

- [ ] Support VERS Debian ranges: `vers:debian/>=1:2.4.1-1`
- [ ] Handle epoch versions correctly: `vers:debian/>=2:1.0.0|<2:2.0.0`
- [ ] Handle debian revisions: `vers:debian/>=2.4.1-1ubuntu1|<2.4.2`
- [ ] Support all VERS operators: `>=`, `<=`, `>`, `<`, `=`, `!=`
- [ ] Pass comprehensive test suite covering edge cases
- [ ] CLI integration: `univers vers contains "vers:debian/>=1:2.4.1" "1:2.5.0-1"`
- [ ] Consistent behavior with existing Debian ecosystem implementation

## Resources

- [VERS Specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst)
- [Univers Python Implementation](https://github.com/aboutcode-org/univers)
- [Debian Policy Manual - Version numbering](https://www.debian.org/doc/debian-policy/ch-controlfields.html#version)
- Existing implementations: [npm VERS](https://github.com/alowayed/go-univers/issues/35), [pypi VERS](https://github.com/alowayed/go-univers/issues/36), [go VERS](https://github.com/alowayed/go-univers/issues/37)
- Pattern reference: `pkg/spec/vers/npm.go`

### PR DESCRIPTION
**Pull request by alowayed**
## Summary

This PR implements VERS (Version Range Specification) support for the Debian ecosystem, addressing issue #55.

- ✅ Adds `debianContains()` and `intervalToDebianRanges()` functions in `pkg/spec/vers/debian.go`
- ✅ Comprehensive test suite with 32 test cases covering all scenarios
- ✅ Supports epoch versions: `vers:debian/>=1:2.4.1-1|<1:3.0.0`
- ✅ Supports revision versions: `vers:debian/>=2.4.1-1ubuntu1|<2.4.2`
- ✅ All VERS operators: `>=`, `<=`, `>`, `<`, `=`, `!=`
- ✅ CLI integration: `univers vers contains "vers:debian/>=1:2.4.1" "1:2.5.0-1"`
- ✅ Uses comma-separated constraints like rpm, gem and cargo for consistency

### Epoch versions
```bash
univers vers contains "vers:debian/>=1:2.4.1-1" "1:2.5.0-1"       # → true
univers vers contains "vers:debian/>=2:1.0.0|<=2:2.0.0" "1:5.0.0" # → false
```

### Debian revision versions (including Ubuntu)
```bash
univers vers contains "vers:debian/>=2.4.1-1ubuntu1" "2.4.1-2ubuntu1" # → true
```

### Complex ranges with epochs and revisions
```bash
univers vers contains "vers:debian/>=1:2.4.1-1|<1:3.0.0|!=1:2.5.0" "1:2.6.0-1" # → true
```

## Test Plan

- [x] All 32 test cases pass including epoch, revision, and error scenarios
- [x] CLI integration verified with epoch and Ubuntu revision versions
- [x] Full test suite passes (no regressions)
- [x] Code formatting and linting passes
- [x] Follows established VERS implementation patterns

## References

- Fixes #55
- Follows patterns established in npm (#35), pypi (#36), go (#37) VERS implementations
- [VERS Specification](https://github.com/package-url/purl-spec/blob/main/VERSION-RANGE-SPEC.rst)
- [Debian Policy Manual - Version numbering](https://www.debian.org/doc/debian-policy/ch-controlfields.html#version)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: pkg/spec/vers/debian.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		return []string{fmt.Sprintf("=%s", interval.exact)}
+	}
+
+	// Exclusions are handled separately, not as Debian ranges
+	if interval.exclude != "" {
+		return []string{} // Return empty - excludes handled in contains function
+	}
+
+	// Handle regular intervals with bounds
+	var parts []string
```
[the remark this answers is not in the document]
Comment (alowayed) [reply pkg/spec/vers/debian.go:29]: No
